fedimint-client/src/lib.rs

```diff
@@ -426,14 +426,40 @@ impl Client {
     }
 
     pub async fn await_tx_accepted(&self, operation_id: OperationId, txid: TransactionId) {
-        self.inner.transaction_update_stream(operation_id).await.filter(|tx_update| std::future::ready(matches!(tx_update.state, TxSubmissionStates::Accepted {txid: event_txid} if event_txid == txid))).next_or_pending().await;
+        self.inner
+            .transaction_update_stream(operation_id)
+            .await
+            .filter(|tx_update| {
+                std::future::ready(matches!(
+                    tx_update.state,
+                    TxSubmissionStates::Accepted {txid: event_txid} if event_txid == txid
+                ))
+            })
+            .next_or_pending()
+            .await;
+    }
+
+    /// Returns the instance id of the first module of the given kind. The
+    /// primary module will always be returned before any other modules (which
+    /// themselves are ordered by their instance id).
+    pub fn get_first_instance(&self, module_kind: &ModuleKind) -> Option<ModuleInstanceId> {
+        if &self.inner.primary_module_kind == module_kind {
+            return Some(self.inner.primary_module_instance);
+        }
+
+        self.inner
+            .modules
+            .iter_modules()
+            .find(|(_, kind, _module)| *kind == module_kind)
+            .map(|(instance_id, _, _)| instance_id)
     }
 }
 
 struct ClientInner {
     db: Database,
     primary_module: DynPrimaryClientModule,
     primary_module_instance: ModuleInstanceId,
+    primary_module_kind: ModuleKind,
```

Comment:
Could be method? :)

Reply:
Would be made obsolete by https://github.com/fedimint/fedimint/issues/2319 too